adafruit_bluefruitspi.py

```diff
@@ -0,0 +1,220 @@
+# The MIT License (MIT)
+#
+# Copyright (c) 2018 Kevin Townsend for Adafruit_Industries
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+"""
+`adafruit_bluefruitspi`
+====================================================
+
+Helper class to work with the Adafruit Bluefruit LE SPI friend breakout.
+
+* Author(s): Kevin Townsend
+
+Implementation Notes
+--------------------
+
+**Hardware:**
+
+"* `Adafruit Bluefruit LE SPI Friend <https://www.adafruit.com/product/2633>`_"
+
+**Software and Dependencies:**
+
+* Adafruit CircuitPython firmware for the supported boards:
+  https://github.com/adafruit/circuitpython/releases
+
+* Adafruit's Bus Device library: https://github.com/adafruit/Adafruit_CircuitPython_BusDevice
+"""
+
+# imports
+
+__version__ = "0.0.0-auto.0"
+__repo__ = "https://github.com/adafruit/Adafruit_CircuitPython_BluefruitSPI.git"
+
+import time
+import digitalio
+from adafruit_bus_device.spi_device import SPIDevice
+from micropython import const
+try:
+    from struct import pack_into, unpack
+except ImportError:
+    from ustruct import pack_into, unpack
+
+
+class MsgType:  #pylint: disable=too-few-public-methods,bad-whitespace
+    """An enum-like class representing the possible message types.
+    Possible values are:
+    - ``MsgType.COMMAND``
+    - ``MsgType.RESPONSE``
+    - ``MsgType.ALERT``
+    - ``MsgType.ERROR``
+    """
+    COMMAND   = const(0x10)  # Command message
+    RESPONSE  = const(0x20)  # Response message
+    ALERT     = const(0x40)  # Alert message
+    ERROR     = const(0x80)  # Error message
```

Comment:
I'd suggest removing these consts. It's a micropythonism that doesn't do anything in this case. (It only works on globals with names that start with _)

Reply:
im redoing these as all plain consts since we only use them internally